### PR TITLE
Convert test/services tests to rspec

### PR DESCRIPTION
**Pull request by sihugh**
I've replaced a number of tests with simpler versions that stub the search API call instead of `Services.search`, and therefore assert more fully that parameters haven't changed. It makes them more consistent with the way do things elsewhere too.

I think Github's tripped over the file moves a little, but they show OK in local changes.

https://trello.com/c/EzMBK0uf/1426-convert-services-tests-to-rspec

https://github.com/alphagov/collections/issues/2259

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: spec/services/location_lookup_service_spec.rb [hunk 1]
```diff
@@ -1,4 +1,3 @@
-require "test_helper"
 require "gds_api/test_helpers/mapit"
 
 describe LocationLookupService do
```
Comment (hannako) [inline spec/services/location_lookup_service_spec.rb:3]: `RSpec.describe` 
The parser should do it but doesn't, and I didn't spend the time to figure out why sorry!

File: spec/services/location_lookup_service_spec.rb [hunk 2]
```diff
@@ -59,27 +58,27 @@
...
 
     it "returns an error if the postcode is not valid" do
       invalid_postcode = "hello"
       stub_mapit_does_not_have_a_bad_postcode(invalid_postcode)
 
-      assert_equal([], described_class.new(invalid_postcode).data)
-      assert_match(invalid_postcode, described_class.new(invalid_postcode).error[:message])
-      assert(described_class.new(invalid_postcode).invalid_postcode?)
+      expect(described_class.new(invalid_postcode).data).to eq([])
+      expect(described_class.new(invalid_postcode).error.[](:message)).to match(invalid_postcode)
```
Comment (hannako) [inline spec/services/location_lookup_service_spec.rb:80]: one of those weird `.[]`

File: spec/services/most_popular_content_spec.rb [hunk 3]
```diff
@@ -1,71 +1,38 @@
...
+        filter_part_of_taxonomy_tree: [taxon_content_id],
+        filter_content_store_document_type: document_types,
+      }
 
-    it "scopes the results to the current taxon" do
-      assert_includes_params(filter_part_of_taxonomy_tree: Array(taxon_content_id)) do
-        most_popular_content.fetch
-      end
-    end
+      stub_search(params: params, body: search_results)
```
Comment (hannako) [inline spec/services/most_popular_content_spec.rb:32]: Oh that's a much better idea

File: spec/services/supergroup_sections_spec.rb [hunk 4]
```diff
@@ -1,26 +1,30 @@
...
+        Supergroups::PolicyAndEngagement => section_tagged_content_list("case_study"),
+        Supergroups::Services => section_tagged_content_list("form"),
+        Supergroups::GuidanceAndRegulation => section_tagged_content_list("guide"),
+        Supergroups::NewsAndCommunications => [],
+        Supergroups::Transparency => [],
+        Supergroups::ResearchAndStatistics => [],
+      }
+
+      stub_config.each do |klass, results|
+        allow_any_instance_of(klass).to receive(:tagged_content).and_return(results)
```
Comment (hannako) [inline spec/services/supergroup_sections_spec.rb:22]: It's missing off a `.with(taxon_id)` bit. Is that needed?

File: spec/services/supergroup_sections_spec.rb [hunk 5]
```diff
@@ -1,26 +1,30 @@
...
+      end
     end
 
-    it "returns a list of sections with tagged content" do
-      assert 5, @sections.length
+    it "returns a list of supergroup sections" do
+      expect(sections.length).to be 6
```
Comment (hannako) [inline spec/services/supergroup_sections_spec.rb:27]: How come it was 5 before but now it's 6?

Comment (sihugh) [reply spec/services/supergroup_sections_spec.rb:27]: It's counting the number of supergroups of which there are 6. The previous thing was asserting that `5` was not `nil` and the message to display was `@sections.length`. I'm taking this as an error that I'm correcting.

File: spec/services/location_lookup_service_spec.rb (hunk 2 above)
Comment (sihugh) [reply spec/services/location_lookup_service_spec.rb:80]: Boo. Thanks!

File: spec/services/supergroup_sections_spec.rb (hunk 4 above)
Comment (sihugh) [reply spec/services/supergroup_sections_spec.rb]: Seems legit 👍

File: spec/services/supergroup_sections_spec.rb (hunk 5 above)
Comment (sihugh) [reply spec/services/supergroup_sections_spec.rb:30]: (So to be clear, the previous assertion wasn't testing what it thought it was testing)

**Review by hannako — APPROVED** [review]

**Pull request merged**